### ISSUE TITLE
Question about min_sample_size (wav2vec)

### ISSUE THREAD

**Issue by karol-nowakowski** [pytorch/fairseq#2952]
In the pretraining [configuration](https://github.com/pytorch/fairseq/blob/master/examples/wav2vec/config/pretraining/wav2vec2_base_librispeech.yaml) for wav2vec2.0, the minimum sample size is set to 32000. What was the motivation behind removing samples shorter than 2 seconds? Did you experiment with smaller values, such as 16000 frames?
 I am trying to pretrain a model from scratch on small amount of data (ca. 150h). Since many clips in my dataset are short, maintaining `min_sample_size=32000` would result in the exclusion of around 10% of the data, hence the question.
 Or should I rather preprocess the data by merging multiple short clips to create longer ones?
 I would appreciate if you could share your experiences in this matter.

**Comment by alexeib** [comment]
if you have very short examples. the convolution in the feature encoder will fail (since there is no padding). also since we to contrastive training and sample negatives from the same utterance, you need at least 2 timesteps after feature extractor, which means you need at least 640 frames.

 2s was pretty arbitrary, you could go shorter as well if you wish - as long as its longer than 640 frames

**Comment by karol-nowakowski** [comment]
Thanks a lot!
